webapp-client/src/component/AddCategoryForm.js

```diff
@@ -1,32 +1,73 @@
 import React from 'react';
+import ErrorMessage from './ErrorMessage';
 
 export default class AddCategoryForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      hasError: false,
+    };
+    this.createCategory = this.createCategory.bind(this);
+    this.clearHasError = this.clearHasError.bind(this);
+  }
   // This function will be changed later to directly POST request to db instead of calling addCategory
-  createCategory = (e) => {
+  async createCategory(e) {
     e.preventDefault();
     const category = e.target.elements['add-category'].value;
-    this.props.addCategory(category);
-    e.target.elements['add-category'].value = ''; 
-  };
+    e.target.elements['add-category'].value = '';
+    const request = {
+      method: 'POST',
+      body: JSON.stringify({
+        title: category,
+        userId: 'fake-user-id',
+      }),
+      headers: { 'Content-type': 'application/json' },
+    };
+    try {
+      const response = await fetch(
+        process.env.REACT_APP_API_URL + '/category/createCategory',
+        request
+      );
+      if (response.status !== 200) {
+        throw new Error(response.statusMessage);
```

Comment:
Do you know what the server's error message is? Is it something we actually want to display?

Reply:
Ah we don't actually display it since it's something technical, I'll change it to an empty throw